aws/src/main/python/beam_lambda/lambda_function.py

```diff
@@ -214,114 +215,120 @@
+  -   echo "looping config ..."
+  -   export MAXRAM=$MAX_RAM
+  -   export SIGOPT_CLIENT_ID="$SIGOPT_CLIENT_ID"
+  -   export SIGOPT_DEV_ID="$SIGOPT_DEV_ID"
+  -   export GOOGLE_API_KEY="$GOOGLE_API_KEY"
+  -   echo $MAXRAM
+  -   /tmp/slack.sh "$hello_msg"
+
+  -   s3p=""
+  -   for cf in $CONFIG
+  -    do
+  -      echo "-------------------running $cf----------------------"
+  -      $RUN_SCRIPT
+  -    done
+  -   echo "-------------------running Health Analysis Script----------------------"
+  -   python3 src/main/python/general_analysis/simulation_health_analysis.py
+  -   while IFS="," read -r metric count
+  -   do
+  -      export $metric=$count
+  -   done < RunHealthAnalysis.txt
+  
+  -   curl -H "Authorization:Bearer $SLACK_TOKEN" -F file=@RunHealthAnalysis.txt -F initial_comment="Beam Health Analysis" -F channels="$SLACK_CHANNEL" "https://slack.com/api/files.upload"
+  -   s3glip=""
+  -   if [ "$S3_PUBLISH" = "True" ]
+  -   then
+  -     s3glip="\\n S3 output url ${s3p#","}"
+  -   fi
+  -   bye_msg=$(printf "Run Completed \\n Run Name** $TITLED** \\n Instance ID %s \\n Instance type **%s** \\n Host name **%s** \\n Web browser ** http://%s:8000 ** \\n Region $REGION \\n Batch $UID \\n Branch **$BRANCH** \\n Commit $COMMIT %s \\n Shutdown in $SHUTDOWN_WAIT minutes" $(ec2metadata --instance-id) $(ec2metadata --instance-type) $(ec2metadata --public-hostname) $(ec2metadata --public-hostname) "$s3glip")
+  -   echo "$bye_msg"
+  -   stop_json=$(printf "{
+        \\"command\\":\\"add\\",
+        \\"type\\":\\"beam\\",
+        \\"sheet_id\\":\\"$SHEET_ID\\",
+        \\"run\\":{
+          \\"status\\":\\"Run Completed\\",
+          \\"name\\":\\"$TITLED\\",
+          \\"instance_id\\":\\"%s\\",
+          \\"instance_type\\":\\"%s\\",
+          \\"host_name\\":\\"%s\\",
+          \\"browser\\":\\"http://%s:8000\\",
+          \\"branch\\":\\"$BRANCH\\",
+          \\"commit\\":\\"$RESOLVED_COMMIT\\",
+          \\"data_branch\\":\\"$DATA_BRANCH\\",
+          \\"data_commit\\":\\"$RESOLVED_DATA_COMMIT\\",
+          \\"region\\":\\"$REGION\\",
+          \\"batch\\":\\"$UID\\",
+          \\"s3_link\\":\\"%s\\",
+          \\"max_ram\\":\\"$MAX_RAM\\",
+          \\"profiler_type\\":\\"$PROFILER\\",
+          \\"config_file\\":\\"$CONFIG\\",
+          \\"stacktrace\\":\\"$stacktrace\\",
+          \\"died_actors\\":\\"$actorDied\\",
+          \\"error\\":\\"$error\\",
+          \\"warning\\":\\"$warn\\",
+          \\"sigopt_client_id\\":\\"$SIGOPT_CLIENT_ID\\",
+          \\"sigopt_dev_id\\":\\"$SIGOPT_DEV_ID\\"
+        }
+      }" $(ec2metadata --instance-id) $(ec2metadata --instance-type) $(ec2metadata --public-hostname) $(ec2metadata --public-hostname) "${s3p#","}")
+  -   /tmp/slack.sh "$bye_msg"
+  -   curl -X POST "https://ca4ircx74d.execute-api.us-east-2.amazonaws.com/production/spreadsheet" -H "Content-Type:application/json" --data "$stop_json"
+  -   $END_SCRIPT
+  -   sudo shutdown -h +$SHUTDOWN_WAIT
```

Comment:
Also giving it a second thought, let's keep the automatic shutdown at 8 hours by-default, and at any value if it was set as gradle parameter.

Reply:
done